Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -92,18 +92,18 @@ void ScrollAnchoringController::invalidateAnchorElement()
     frameView().dequeueScrollableAreaForScrollAnchoringUpdate(m_owningScrollableArea);
 }
 
-static IntRect boundingRectForScrollableArea(ScrollableArea& scrollableArea)
+static IntRect boundingRectForScrollableArea(ScrollableArea& scrollableArea, bool shouldIncludeLayoutViewportRect = true)
 {
     if (auto* renderLayerScrollableArea = dynamicDowncast<RenderLayerScrollableArea>(scrollableArea))
         return renderLayerScrollableArea->layer().renderer().absoluteBoundingBoxRect();
 
-    return IntRect(downcast<LocalFrameView>(downcast<ScrollView>(scrollableArea)).layoutViewportRect());
+    return shouldIncludeLayoutViewportRect ? IntRect(downcast<LocalFrameView>(downcast<ScrollView>(scrollableArea)).layoutViewportRect()) : IntRect();
```

Comment:
Why does it return an empty rect instead of the FrameView's bounds or something? I'm also not sure what it means to "include" the layout viewport.

Seems like it would be cleaner to just return the FrameView's bounds, and have the caller deal with the behavior divergence.

Reply:
Renamed this to shouldIncludeFrameViewLocation. Made this only used in `ScrollAnchoringController::computeOffsetFromScrollableArea`.